### DISCUSSION TITLE
How to trigger a Vercel build on content update?

### DISCUSSION THREAD

**Discussion by sebpowell** [sanity-io/sanity#4492] [Sanity Studio Support]
I'm new to Sanity, and trying to familiarise myself with how it works. 

I'm trying to work out how to trigger a build in Vercel when the content is updated in the studio. From what I gather, this requires using the following plugin: https://www.npmjs.com/package/sanity-plugin-vercel-deploy

I installed it, and it works fine locally – I can see the 'deploy' tab on `localhost:3000/studio`. However, it doesn't show up when I access the studio on Vercel.

I was wondering whether it might be a setting I need to enable somewhere, but wasn't able to see anything obvious in the docs. I've left a ticket on the project Github, but wondering if anyone here might have an idea as to where I'm going wrong. 

This is my package.json:

```
{
  "name": "next-sanity-test",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "ts": "tsc --watch --noEmit"
  },
  "dependencies": {
    "@portabletext/react": "^2.0.2",
    "@sanity/block-content-to-markdown": "^0.0.5",
    "@sanity/code-input": "^4.1.1",
    "@sanity/image-url": "^1.0.2",
    "@sanity/vision": "^3.10.1",
    "@types/node": "18.15.0",
    "@types/react": "18.0.28",
    "@types/react-dom": "18.0.11",
    "eslint": "8.35.0",
    "eslint-config-next": "13.2.3",
    "next": "13.2.3",
    "next-sanity": "^4.3.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "sanity": "^3.10.1",
    "sanity-plugin-vercel-deploy": "^3.3.3",
    "typescript": "4.9.5"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.21",
    "prettier": "^2.8.8",
    "sanity-codegen": "^0.9.8",
    "tailwindcss": "^3.2.7"
  }
}
```

And this is my sanity.config.ts file:

```
import { visionTool } from "@sanity/vision";
import { defineConfig } from "sanity";
import { deskTool } from "sanity/desk";
import { codeInput } from "@sanity/code-input";
import { apiVersion, dataset, projectId } from "./sanity/env";
import schemas from "./sanity/schemas";
import { vercelDeployTool } from 'sanity-plugin-vercel-deploy'

export default defineConfig({
  basePath: "/studio",
  projectId,
  dataset,
  schema: {
    types: [...schemas],
  },
  plugins: [
    deskTool(),
    visionTool({ defaultApiVersion: apiVersion }),
    codeInput(),
    vercelDeployTool(),
  ],
});
```

The tab shows on localhost:

![image](https://github.com/sanity-io/sanity/assets/1786366/a018c884-204c-47ed-9dc4-6840fca9795e)

But not on production:
![image](https://github.com/sanity-io/sanity/assets/1786366/6cb8a7c1-f983-4d1c-91e2-b8932a5fd086)

Thank you for your help!

**Comment by basharalidreesi** [comment]


**Reply by sebpowell** [reply]
@basharalidreesi – thanks for the suggestion! It seems to have suddenly started working now, not sure what happened. In any case, I don't think the above was the problem as I'm self-hosting on Vercel. Thank you for the suggestion in any case!
